src/pseudopeople/noise_functions.py

```diff
@@ -28,117 +25,56 @@
     vectorized_choice,
 )
 
+if TYPE_CHECKING:
+    from pseudopeople.dataset import DatasetData
+
```

Comment:
Everything in this file is either:
-removing code that has been moved to a better location
-update function signatures
-updating missingness attribute for a few noise types
-updates are new class in place and now ALL noise functions return None.